src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/JsonPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,233 @@
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.TRAILING)
+                                    .addComponent(numberOfPayloadsLabel)
+                                    .addComponent(jsonFieldLabel))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                                    .addComponent(numberOfPayloadsSpinner)
+                                    .addComponent(jsonFieldScrollPane)));
+
+            layout.setVerticalGroup(layout.createSequentialGroup()
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(numberOfPayloadsLabel)
+                                    .addComponent(numberOfPayloadsSpinner))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(jsonFieldLabel)
+                                    .addComponent(jsonFieldScrollPane)));
+        }
+
+        @Override
+        protected JsonPayloadGenerator getPayloadGenerator() {
+            if (!validate()) {
+                return null;
+            }
+            String j = jsonTextArea.getText();
+            int i = numberOfPayloadsSpinner.getValue();
+            return new JsonPayloadGenerator(j, i);
+        }
+
+        @Override
+        public void init(MessageLocation messageLocation) {
+
+        }
+
+        @Override
+        public JPanel getComponent() {
+            return fieldsPanel;
+        }
+
+        @Override
+        public void setPayloadGeneratorUI(JsonPayloadGeneratorUI payloadGeneratorUI) {
+            oldGenerator = payloadGeneratorUI;
+            numberOfPayloadsSpinner.setValue(payloadGeneratorUI.getPayloadGenerator().getNumberOfPayloads());
+            jsonTextArea.setText(payloadGeneratorUI.getPayloadGenerator().getJson());
+        }
+
+        @Override
+        public JsonPayloadGeneratorUI getPayloadGeneratorUI() {
+            if (oldGenerator != null) {
+                return oldGenerator;
+            }
+
+            return new JsonPayloadGeneratorUI(getPayloadGenerator());
+        }
+
+        @Override
+        public void clear() {
+            oldGenerator = null;
+
+            getJsonTextArea().setText("");
+            numberOfPayloadsSpinner.setValue(1);
+        }
+
+        @Override
+        public boolean validate() {
+            if (oldGenerator != null &&
+                    getJsonTextArea().getText().equals(oldGenerator.getPayloadGenerator().getJson()) &&
+                    numberOfPayloadsSpinner.getValue() == oldGenerator.getPayloadGenerator().getNumberOfPayloads()) {
+                return true;
+            }
+
+            if (getJsonTextArea().getText().isEmpty()) return false;
```

Comment:
- Could do `getJsonTextArea().getDocument().getLength() == 0` to avoid creating the string.
 - For consistency the if should have braces.